src/index.js

```diff
@@ -11,9 +11,24 @@ export function AggregateError(message: string, innerError?: Error, skipIfAlread
       return innerError;
     }
 
-    if (innerError.stack) {
-      message += `\n------------------------------------------------\ninner error: ${innerError.stack}`;
+    message += '\n------------------------------------------------\nInner Error:\n';
+
+    if (typeof(innerError) === 'string' ) {
+      message += `Message: ${innerError}`;
+    } else {
+      if (innerError.message) {
+        message += `Message: ${innerError.message}`;
+      } else {
+        message += `Unknown Inner Error Type. Displaying Inner Error as JSON:\n ${JSON.stringify(innerError, null, '  ')}`;
+      }
+
+      if (innerError.stack) {
+        message += `\nInner Error Stack:\n${innerError.stack}`;
+        message += '\nEnd Inner Error Stack';
+      }
     }
+
+    message += '\n------------------------------------------------\n';
```

Comment:
One fix before we merge this. Let's factor out the `\n---etc.` bit into a const so it can be used in both places.